### PR TITLE
add official support to Python 3.13

### PR DESCRIPTION
**Pull request by emdneto**
# Description

py313 issues
- [X] sqlalchemy -- https://github.com/open-telemetry/opentelemetry-python-contrib/pull/3139
- [x] grpc -- https://github.com/open-telemetry/opentelemetry-python-contrib/pull/3140
- [x] asyncpg -- https://github.com/open-telemetry/opentelemetry-python-contrib/pull/3146
- [x] Celery -- https://github.com/open-telemetry/opentelemetry-python-contrib/pull/3145
- [x] psycopg -- https://github.com/open-telemetry/opentelemetry-python-contrib/pull/3147
- [x] psycopg2 -- https://github.com/open-telemetry/opentelemetry-python-contrib/pull/3147
- [x] aiopg -- https://github.com/open-telemetry/opentelemetry-python-contrib/pull/3147
- [x] falcon - https://github.com/open-telemetry/opentelemetry-python-contrib/pull/3086
- [x] cassandra -- https://github.com/open-telemetry/opentelemetry-python-contrib/pull/3152
- [x] prometheus-remote-write -- https://github.com/open-telemetry/opentelemetry-python-contrib/pull/3157  
- [X] httpx -- fixed in the PR itself (tox.ini)
- [x] django -- fixed in the PR itself (tox.ini)

The only instrumentation we are not running py313 is pyramid. Will open a track issue for that:
- [x] pyramid - Disabled py313 for now. See https://github.com/Pylons/pyramid/pull/3767

Closes https://github.com/open-telemetry/opentelemetry-python/issues/4211 
Closes https://github.com/open-telemetry/opentelemetry-python-contrib/pull/2724

### CODE REVIEW

File: .github/workflows/generate_workflows_lib/pyproject.toml [hunk 1]
```diff
@@ -23,6 +23,7 @@ classifiers = [
   "Programming Language :: Python :: 3.11",
   "Programming Language :: Python :: 3.12",
   "Typing :: Typed",
+  "Programming Language :: Python :: 3.13",
```
Comment (Kludex) [inline .github/workflows/generate_workflows_lib/pyproject.toml:26]:
```suggestion
  "Programming Language :: Python :: 3.13",
  "Typing :: Typed",
```

**Review by Kludex — APPROVED** [review]

File: .github/workflows/generate_workflows_lib/pyproject.toml (hunk 1 above)
Comment (emdneto) [reply .github/workflows/generate_workflows_lib/pyproject.toml]: oops, sed failed

**Review by xrmx — APPROVED** [review]

File: CHANGELOG.md [hunk 2]
```diff
@@ -19,7 +19,8 @@ and this project adheres to [Semantic Versioning](https://semver.org/spec/v2.0.0
   ([#3111](https://github.com/open-telemetry/opentelemetry-python-contrib/pull/3111))
 - `opentelemetry-instrumentation-falcon` add support version to v4
   ([#3086](https://github.com/open-telemetry/opentelemetry-python-contrib/pull/3086))
-
+- add python 3.13 to tox
+  ([#3134](https://github.com/open-telemetry/opentelemetry-python-contrib/pull/3134))
```
Comment (Kludex) [inline CHANGELOG.md:23]: I think the point is more "Officially support Python 3.13" rather than the tox setup here.

**Review by Kludex — CHANGES_REQUESTED** [review]

File: CHANGELOG.md [hunk 3]
```diff
@@ -19,7 +19,8 @@ and this project adheres to [Semantic Versioning](https://semver.org/spec/v2.0.0
   ([#3111](https://github.com/open-telemetry/opentelemetry-python-contrib/pull/3111))
 - `opentelemetry-instrumentation-falcon` add support version to v4
   ([#3086](https://github.com/open-telemetry/opentelemetry-python-contrib/pull/3086))
-
+- add python 3.13 to tox
```
Comment (emdneto) [inline CHANGELOG.md:22]:
```suggestion
- add support to python 3.13
```

File: CHANGELOG.md (hunk 2 above)
Comment (emdneto) [reply CHANGELOG.md]: I'm fine changing the wording here

**Comment by emdneto** [comment]
> nit with context: I'm trying to figure out where I can contribute without asking, and I feel like I could do what I'm asking here. However, not sure it is my place or not.
> 
> Is this also implicitly dropping 3.8? If so, maybe the changelog and PR title could be
> 
> "Support Python 3.13 and drop testing of 3.8"
> 
> I'm not sure what the "official" word means, I guess we meant we didn't prevent people from using it?

@codefromthecrypt, thanks for your concern. It's important to note we are not dropping support for python3.8. We do still support 3.8 until 3.13. 

"Officially" supporting a Python version is defined [here](https://github.com/open-telemetry/opentelemetry-python?tab=readme-ov-file#python-version-support), and it basically means: "Now we are running tests at CI using this python version and we do care about issues of our packages when running with this version. We didn't prevent people from using it before, but now users have some guarantee that packages run fine with that version, and we support it. I hope that it makes sense to you, but let me know if there's something we can improve on the wording here.

File: tox.ini [hunk 4]
```diff
@@ -141,13 +143,13 @@ envlist =
     ; 0: Flask ==2.1.3 Werkzeug <3.0.0
     ; 1: Flask ==2.2.0 Werkzeug <3.0.0
     ; 2: Flask >=3.0.0 Werkzeug >=3.0.0
-    py3{8,9,10,11,12}-test-instrumentation-flask-{0,1}
-    py3{8,9,10,11,12}-test-instrumentation-flask-{2}
+    py3{8,9,10,11,12,13}-test-instrumentation-flask-{0,1,2}
+    ; py3{8,9,10,11,12}-test-instrumentation-flask-{2}
```
Comment (emdneto) [inline tox.ini:147]:
```suggestion
```

**Review by codefromthecrypt — APPROVED** [review]
thanks for the help @emdneto. This in mind, I don't see any changes needed here.

**Review by Kludex — APPROVED** [review]

**Pull request merged**